colcon_core/environment/__init__.py

```diff
@@ -77,12 +77,17 @@ def create_environment_scripts(
       by :function:`create_environment_hooks` are used
     :param list additional_hooks: Any additional hooks which should be
       referenced by the generated scripts
+    :returns: iterable of generated file paths
+    :rtype: Iterable
     """
     prefix_path = Path(args.install_base)
-    create_environment_scripts_only(
+    files = []
+    files += create_environment_scripts_only(
         prefix_path, pkg, default_hooks=default_hooks,
         additional_hooks=additional_hooks)
-    create_file_with_runtime_dependencies(prefix_path, pkg)
+    files.append(
+        create_file_with_runtime_dependencies(prefix_path, pkg))
```

Comment:
Why use `files +=` for the first function and `files.append()` for the second? Any answer will suffice, it's not a blocking concern.

Reply:
The `+=` _could_ be changed to `files.extend()` instead, but `.append()` is used to add a single item, where `create_environment_scripts_only` is a list of items. I don't think there's an operator-based equivelant to `.append()` short of creating a temporary list of a single item and using `+=`.

Reply:
> Is it worth cataloging the disallowing of `None` for a future 3.0 ShellExtensionPoint in an issue or other persistent place or do we trust that we'll find the comments when the time comes to touch this all again?

I've actually spotted some other comments like this one that call for follow-ups to deprecations. We should probably make a pass at some point. Maybe we should file a single tracking issue for all of them.